packages/client-node/src/connection/node_base_connection.ts

```diff
@@ -258,19 +265,28 @@ export abstract class NodeBaseConnection
   protected buildRequestHeaders(
     params?: BaseQueryParams,
   ): Http.OutgoingHttpHeaders {
-    if (this.params.set_basic_auth_header) {
+    if (isJWTAuth(params?.auth)) {
       return {
         ...this.defaultHeaders,
-        Authorization:
-          params?.auth !== undefined
-            ? `Basic ${Buffer.from(`${params.auth.username}:${params.auth.password}`).toString('base64')}`
-            : this.defaultAuthHeader,
+        Authorization: `Bearer ${params.auth.access_token}`,
       }
-    } else {
-      return {
-        ...this.defaultHeaders,
+    }
+    if (this.params.set_basic_auth_header) {
+      if (isCredentialsAuth(params?.auth)) {
+        return {
+          ...this.defaultHeaders,
+          Authorization: `Basic ${Buffer.from(`${params.auth.username}:${params.auth.password}`).toString('base64')}`,
+        }
+      } else {
```

Comment:
optional: can we move `Auth header` calculation logic to a separate function and test separately? It's hard to reason about the flow with a few levels of `if`
```js
if (isJWTAuth(params?.auth)) {...}
if (this.params.set_basic_auth_header) {
      if (isCredentialsAuth(params?.auth)) { ... } 
      else {...}
}
```